### PR TITLE
Fix timing perfromance and user-agent models

### PR DESCRIPTION
**Pull request by sphinks**
Fixing issue https://github.com/fishtown-analytics/snowplow/issues/40

### CODE REVIEW

**Comment by drewbanin** [comment]
@sphinks I just kicked off the tests and it looks like everything is passing. That's probably because we're not currently testing these "optional" models though 🙃 

This change may break existing projects, but I think it's good to be consistent with Snowplow's field names. Let me give this a quick spin locally to make sure we're in good shape, and then I think we can merge this!

File: models/page_views/optional/snowplow_web_timing_context.sql [hunk 1]
```diff
@@ -17,7 +17,7 @@ web_page_context as (
 prep as (
 
     select
-        wp.page_view_id,
+        wp.id,
```
Comment (drewbanin) [inline models/page_views/optional/snowplow_web_timing_context.sql]: hmm, does the `web_page_context` model have a field called `id`? I think at present, `web_page_context` exposes [two fields](https://github.com/fishtown-analytics/snowplow/blob/274c7969d61a552cdc996d00801d7f43cc9b6c03/macros/adapters/default/page_views/snowplow_web_page_context.sql#L34-L35):
- root_id
- page_view_id

Does this work?

File: models/page_views/optional/snowplow_web_ua_parser_context.sql [hunk 2]
```diff
@@ -18,7 +18,7 @@ prep AS (
 
     SELECT
 
-      wp.page_view_id,
+      wp.id,
```
Comment (drewbanin) [inline models/page_views/optional/snowplow_web_ua_parser_context.sql]: same here -- I don't believe there's an `id` field in the [snowplow_base_web_page_context](https://github.com/fishtown-analytics/snowplow/blob/master/macros/adapters/default/page_views/snowplow_web_page_context.sql#L34-L35) model?

**Review by drewbanin — DISMISSED** [review]
I gave this another look with some fresh eyes and have a couple of quick questions!

**Comment by drewbanin** [comment]
Gosh... this code is kind of a mess.... So these optional models [select directly from the web_page_context base model](https://github.com/fishtown-analytics/snowplow/blob/274c7969d61a552cdc996d00801d7f43cc9b6c03/models/base/snowplow_base_web_page_context.sql). I get it. 

I think that we can indeed select the `id` column (as you have in this PR) or we could switch it to use the [snowplow_web_page_context](https://github.com/fishtown-analytics/snowplow/blob/274c7969d61a552cdc996d00801d7f43cc9b6c03/models/page_views/snowplow_web_page_context.sql) model which includes 
1) renamed fields
2) deduplicated rows

Curious to hear which of these approaches you think is preferable

**Comment by sphinks** [comment]
Let me a dig a little deeper, because I check at my side and for sure I get an error:

> column wp.page_view_id does not exist
  compiled SQL at target/compiled/snowplow/page_views/optional/snowplow_web_timing_context.sql

in case of original code. We'll take a look on you comment, to understand the rootcause.

**Comment by drewbanin** [comment]
Cool, thanks @sphinks! I appreciate your help and insight here :) 

Let me know what you find!

**Comment by sphinks** [comment]
@drewbanin Have reverted rename of fields and switch referencing from base model to preprocessed web context model, to reuse de-duplication and all the rest code. Please, take a look.

**Comment by drewbanin** [comment]
Thanks @sphinks, this LGTM! Will merge this today

**Comment by sphinks** [comment]
@drewbanin Great!

**Pull request merged**